servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServiceTalkHttpAttributesGetter.java

```diff
@@ -117,35 +116,34 @@ private static final class ClientGetter extends ServiceTalkHttpAttributesGetter
         @Override
         @Nullable
         public String getUrlFull(final HttpRequestMetaData request) {
+            String requestTarget = request.requestTarget();
+            if (requestTarget.startsWith("https://") || requestTarget.startsWith("http://")) {
+                // request target is already absolute-form: just return it.
+                return requestTarget;
+            }
+
+            // in this case the request target is most likely origin-form so we need to convert it to absolute-form.
             HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
             if (effectiveHostAndPort == null) {
+                // we cant create the authority so we must just return.
                 return null;
             }
             String scheme = request.scheme();
             if (scheme == null) {
-                scheme = HTTP_SCHEME;
+                scheme = effectiveHostAndPort.port() == 443 ? HTTPS_SCHEME : HTTP_SCHEME;
             }
-            String requestTarget = request.requestTarget();
-            StringBuilder sb = new StringBuilder(
-                    scheme.length() + 3 +
-                    effectiveHostAndPort.hostName().length() +
-                    ((effectiveHostAndPort.port()) >= 0 ? 5 : 0) +
-                    requestTarget.length());
-            sb.append(scheme)
-                    .append("://")
-                    .append(effectiveHostAndPort.hostName());
-            if (effectiveHostAndPort.port() >= 0) {
-                sb.append(':')
-                        .append(effectiveHostAndPort.port());
+            String authority = effectiveHostAndPort.hostName();
+            if (!isDefaultPort(scheme, effectiveHostAndPort.port())) {
+                authority = authority + ':' + effectiveHostAndPort.port();
             }
-            sb.append(requestTarget);
-            return sb.toString();
+            String authoritySeparator = requestTarget.startsWith("/") ? "" : "/";
+            return scheme + "://" + authority + authoritySeparator + requestTarget;
```

Comment:
Could you remind me what we decided on our last discussion: should we use `path()` or `requestTarget()`?

Reply:
I see `io.opentelemetry.instrumentation.api.semconv.http.HttpClientAttributesExtractor` has `stripSensitiveData` method, but it's private. They simply hardcoded some well-known parameters, like `AWSAccessKeyId` or `Signature` 😕